service/storage_proxy.cc

```diff
@@ -3207,14 +3207,14 @@ storage_proxy::create_write_response_handler_helper(schema_ptr s, const dht::tok
         // The idea is that if we have over maxHintsInProgress hints in flight, this is probably due to
         // a small number of nodes causing problems, so we should avoid shutting down writes completely to
         // healthy nodes.  Any node with no hintsInProgress is considered healthy.
-        throw overloaded_exception(_hints_manager.size_of_hints_in_progress());
+        return boost::outcome_v2::failure(overloaded_exception(_hints_manager.size_of_hints_in_progress()));
```

Comment:
> because the exception, converted into coordinator_exception_container
> can be then converted into both into a successful response_id_type
> as well as into a failure.

How can `coordinator_exception_container` be converted into `response_id_type` ?? How does that work?

Reply:
`response_id_type` is just an `uint64_t` and it seems like the `coordinator_exception_container`'s bool operator is used for the conversion so we get 1 or 0